daemon/endpoint.go

```diff
@@ -429,7 +425,11 @@ func (h *patchEndpointID) Handle(params PatchEndpointIDParams) middleware.Respon
 			reason = "Waiting on endpoint initial program regeneration while handling API PATCH"
 		}
 	}
-	ep.Mutex.Unlock()
+
+	// If endpoint is disconnected, the request should fail
```

Comment:
This seems wrong, we should be unlocking here.

Reply:
done